mlflow/utils/environment.py

```diff
@@ -230,6 +231,18 @@ def _is_mlflow_requirement(requirement_string):
     except InvalidRequirement:
         # A local file path or URL falls into this branch.
 
+        # `Requirement` throws an `InvalidRequirement` exception if `requirement_string` contains per-requirement options (ex: package hashes)
+        # GitHub issue: https://github.com/pypa/packaging/issues/488
+        # Per-requirement-option spec: https://pip.pypa.io/en/stable/reference/requirements-file-format/#per-requirement-options
+        if "--" in requirement_string:
+            match = re.search("^(.*?)((?:\\s+--\\w+=\\S+)*)$", requirement_string)
```

Comment:
```suggestion
            match = re.search(r"^(.*?)((?:\s+--\w+=\S+)*)$", requirement_string)
```

Can we use a raw string?

Reply:
Found a function used in pip for processing a requirement string:
https://github.com/pypa/pip/blob/b392833a0f1cff1bbee1ac6dbe0270cccdd0c11f/src/pip/_internal/req/req_file.py#L400

```python
from typing import Tuple


def break_args_options(line: str) -> Tuple[str, str]:
    """Break up the line into an args and options string.  We only want to shlex
    (and then optparse) the options, not the args.  args can contain markers
    which are corrupted by shlex.
    """
    tokens = line.split(" ")
    args = []
    options = tokens[:]
    for token in tokens:
        if token.startswith("-") or token.startswith("--"):
            break
        else:
            args.append(token)
            options.pop(0)
    return " ".join(args), " ".join(options)


print(break_args_options("mlflow"))
print(break_args_options("mlflow==1.2.3"))
print(break_args_options("-r reqs.txt"))
print(break_args_options("mlflow==1.2.3 --hash=foo"))
```

output:

```
('mlflow', '')
('mlflow==1.2.3', '')
('', '-r reqs.txt')
('mlflow==1.2.3', '--hash=foo')
```

Can we do something similar?

Reply:
Updated code to `split`-based solution